tailscale/resource_dns_configuration.go

```diff
@@ -0,0 +1,257 @@
+		ReadContext:   resourceDNSConfigurationRead,
+		CreateContext: resourceDNSConfigurationCreate,
+		UpdateContext: resourceDNSConfigurationUpdate,
+		DeleteContext: resourceDNSConfigurationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"nameservers": {
+				Description: "Set the nameservers used by devices on your network to resolve DNS queries.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"address": {
+							Description: "The nameserver's IPv4 or IPv6 address",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"use_with_exit_node": {
+							Description: "This name server will continue to be used when an exit node is selected (requires Tailscale v1.88 or later).",
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Default:     false,
+						},
+					},
+				},
+			},
+			"split_dns": {
+				Description: "Set the nameservers used by devices on your network to resolve DNS queries on specific domains (requires Tailscale v1.8 or later).",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"domain": {
+							Description: "The nameservers will be used only for this domain.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"nameservers": {
+							Description: "Set the nameservers used by devices on your network to resolve DNS queries.",
+							Type:        schema.TypeList,
+							Required:    true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"address": {
+										Description: "The nameserver's IPv4 or IPv6 address.",
+										Type:        schema.TypeString,
+										Required:    true,
+									},
+									"use_with_exit_node": {
+										Description: "This name server will continue to be used when an exit node is selected (requires Tailscale v1.88 or later).",
+										Type:        schema.TypeBool,
+										Optional:    true,
+										Default:     false,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"search_paths": {
+				Description: "Additional search domains. When MagicDNS is on, the tailnet domain is automatically included as the first search domain.",
+				Type:        schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional: true,
+			},
+			"override_local_dns": {
+				Description: "When enabled, use the configured DNS servers to resolve names outside the tailnet. When disabled, devices will prefer their local DNS configuration.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+			},
+			"disable_magic_dns": {
```

Comment:
Is there a reason we have this flipped from the API?

Reply:
Good question. I was just going by what was in the design. @mzbenami thoughts?